checkfiles/checkfiles.py

```diff
@@ -183,6 +185,258 @@ def check_format(encValData, job, path):
+                if len(signatures_for_comparison) == 0:
+                    for entry in signatures_no_barcode_set:
+                        signatures_for_comparison.add(entry + 'mixed:')
+
+            else:
+                signatures_for_comparison = signatures_set
+
+        conflicts = []
+        for unique_signature in signatures_for_comparison:
+            query = '/' + unique_signature + '?format=json'
+            try:
+                r = session.get(urljoin(url, query))
+            except requests.exceptions.RequestException as e:  # This is the correct syntax
+                errors['lookup_for_fastq_signature'] = 'Network error occured, while looking for ' + \
+                                                       'fastq signatures conflict on the portal. ' + \
+                                                       str(e) + \
+                    ' Gathered information about the file was: {}.'.format(str(result))
+            else:
+                response = r.json()
+                if response is not None and 'File' in response['@type']:
+                    uniqueness_flag = False
+                    conflicts.append(
+                        'specified unique identifier {} '.format(unique_signature) +
+                        'is conflicting with identifier of reads from ' +
+                        'file {}.'.format(response['accession']))
+        if uniqueness_flag is True:
+            result['fastq_signature'] = sorted(list(signatures_for_comparison))
+        else:
+            errors['not_unique_flowcell_details'] = str(conflicts.append(
+                ' Gathered information about the file was: {}.'.format(str(result))))
+
+
+def process_barcodes(signatures_set):
+    set_to_return = set()
+    flowcells_dict = {}
+    for entry in signatures_set:
+        arr = entry.split(':')
+        f = arr[0]
+        l = arr[1]
+        r = arr[2]
+        b = arr[3]
+        if (f, l, r) not in flowcells_dict:
+            flowcells_dict[(f, l, r)] = {}
+        if b not in flowcells_dict[(f, l, r)]:
+            flowcells_dict[(f, l, r)][b] = 0
+        flowcells_dict[(f, l, r)][b] += 1
+    for key in flowcells_dict.keys():
+        barcodes_dict = flowcells_dict[key]
+        total = 0
+        for b in barcodes_dict.keys():
+            total += barcodes_dict[b]
+        for b in barcodes_dict.keys():
+            if ((float(total)/float(barcodes_dict[b])) < 100):
+                set_to_return.add(key[0] + ':' +
+                                  key[1] + ':' +
+                                  key[2] + ':' +
+                                  b + ':')
+    return set_to_return
+
+def process_read_lengths(read_lengths_dict,
+                         lengths_list,
+                         submitted_read_length,
+                         read_count,
+                         threshold_percentage,
+                         errors_to_report,
+                         result):
+    expected_read_lengths = [
```

Comment:
this whole bit is pretty clunky too.
you have a  histogram of read lengths and a submitted_length +/- 2.

You want something like a dictionary slice of read_lengths_dict for the 5 keys.
[ number for length, number in read_length_dict.items() if submitted_length -2 <= length <= submitted_length+2 ]
and then sum that array.